File: POLIMORFISMO/ejercicio2/Videojuego.java
package ejercicio2;

import java.util.Scanner;

public class Videojuego {
    // atributos
    private String nombre;
    private String plataforma;
    private int cantidadJugadores;

    // Constructor por defecto
    public Videojuego() {
        this.nombre = "";
        this.plataforma = "";
        this.cantidadJugadores = 0;
    }

    // Constructor con nombre y plataforma
    public Videojuego(String nombre, String plataforma) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = 0;
    }

    // Constructor con todos los atributos
    public Videojuego(String nombre, String plataforma, int cantidadJugadores) {
        this.nombre = nombre;
        this.plataforma = plataforma;
        this.cantidadJugadores = cantidadJugadores;
    }

    // Sobrecarga método agregarJugadores: agrega solo un jugador
    public void agregarJugadores() {
        this.cantidadJugadores++;
    }

    // agrega cantidad ingresada por teclado
    public void agregarJugadores(Scanner sc) {
        System.out.print("Ingrese la cantidad de jugadores a agregar: ");
        int cantidad = sc.nextInt();
        this.cantidadJugadores += cantidad;
    }

    // Getters para mostrar información
    public String getNombre() {
        return nombre;
    }

    public String getPlataforma() {
        return plataforma;
    }

    public int getCantidadJugadores() {
        return cantidadJugadores;
    }

    @Override
    public String toString() {
        return "Videojuego[" +
                "nombre = " + nombre +
                ", plataforma = " + plataforma +
                ", cantidadJugadores = " + cantidadJugadores +
                "]";
    }
}

File: POLIMORFISMO/ejercicio2/videojuego.py
# Clase Videojuego con sobrecarga de constructores y métodos
class Videojuego:
	def __init__(self, nombre=None, plataforma=None, cantidad_jugadores=1):
		if nombre is None and plataforma is None:
			self.nombre = "Desconocido"
			self.plataforma = "Desconocida"
			self.cantidad_jugadores = 1
		elif plataforma is None:
			self.nombre = nombre
			self.plataforma = "Desconocida"
			self.cantidad_jugadores = 1
		else:
			self.nombre = nombre
			self.plataforma = plataforma
			self.cantidad_jugadores = cantidad_jugadores

	# Sobrecarga de agregarJugadores
	def agregarJugadores(self, cantidad=None):
		if cantidad is None:
			self.cantidad_jugadores += 1
			print(f"Se agregó 1 jugador. Total: {self.cantidad_jugadores}")
		else:
			try:
				cantidad = int(cantidad)
				self.cantidad_jugadores += cantidad
				print(f"Se agregaron {cantidad} jugadores. Total: {self.cantidad_jugadores}")
			except ValueError:
				print("Cantidad inválida.")

	def __str__(self):
		return f"Videojuego: {self.nombre}, Plataforma: {self.plataforma}, Jugadores: {self.cantidad_jugadores}"

# programa principal

v1 = Videojuego("FIFA 23", "PS5", 2)
v2 = Videojuego("Minecraft")
v3 = Videojuego()
print(v1)
print(v2)
print(v3)
v1.agregarJugadores()
v2.agregarJugadores(3)
# Para agregar por teclado
cantidad = input("¿Cuántos jugadores agregar a Minecraft?: ")
v2.agregarJugadores(cantidad)
print(v2)
